### PR TITLE
Problem: Missing a demo for unreal 5.1 (fix cronos-labs/play-unreal-plugin#253)

### PR DESCRIPTION
**Pull request by damoncro**
It is a default third character C++ template project without startup content, with wallet connect example.

### CODE REVIEW

File: Metaverse/Config/DefaultEngine.ini [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+[/Script/Engine.Engine]
++ActiveGameNameRedirects=(OldGameName="TP_ThirdPerson",NewGameName="/Script/Metaverse")
++ActiveGameNameRedirects=(OldGameName="/Script/TP_ThirdPerson",NewGameName="/Script/Metaverse")
++ActiveClassRedirects=(OldClassName="TP_ThirdPersonGameMode",NewClassName="MetaverseGameMode")
++ActiveClassRedirects=(OldClassName="TP_ThirdPersonCharacter",NewClassName="MetaverseCharacter")
+
+[/Script/AndroidFileServerEditor.AndroidFileServerRuntimeSettings]
+bEnablePlugin=True
+bAllowNetworkConnection=True
+SecurityToken=CCEED369EB42CBA717F8528399C6F4E9
```
Comment (tomtau) [inline Metaverse/Config/DefaultEngine.ini:60]: I guess this is just dev-only token?

File: Metaverse/Source/Metaverse.Target.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright Epic Games, Inc. All Rights Reserved.
```
Comment (tomtau) [inline Metaverse/Source/Metaverse.Target.cs:1]: is this copyright header correct for the project?

**Review by tomtau — COMMENTED** [review]
I guess all the assets are the Unreal ones (i.e. anyone is free to use them within UE)?

**Comment by tomtau** [comment]
@rarcif

File: Metaverse/Config/DefaultEngine.ini (hunk 1 above)
Comment (rarcifa) [reply Metaverse/Config/DefaultEngine.ini:60]: maybe SecurityToken=SECURITY_TOKEN and pass the token as env variable?

File: Metaverse/Config/DefaultGame.ini [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[/Script/EngineSettings.GeneralProjectSettings]
+ProjectID=D914E5863F4885D22EC07081DBEBD81C
```
Comment (rarcifa) [inline Metaverse/Config/DefaultGame.ini:2]: maybe ProjectID=PROJECT_ID and pass the token as env variable?

File: Metaverse/Source/Metaverse/MetaverseCharacter.cpp [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+//////////////////////////////////////////////////////////////////////////
+// Input
+
+void AMetaverseCharacter::SetupPlayerInputComponent(class UInputComponent* PlayerInputComponent)
+{
+	// Set up action bindings
+	if (UEnhancedInputComponent* EnhancedInputComponent = CastChecked<UEnhancedInputComponent>(PlayerInputComponent)) {
+		
+		//Jumping
```
Comment (rarcifa) [inline Metaverse/Source/Metaverse/MetaverseCharacter.cpp:77]: space // Jumping

File: Metaverse/Source/Metaverse/MetaverseCharacter.cpp [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+		EnhancedInputComponent->BindAction(JumpAction, ETriggerEvent::Triggered, this, &ACharacter::Jump);
+		EnhancedInputComponent->BindAction(JumpAction, ETriggerEvent::Completed, this, &ACharacter::StopJumping);
+
+		//Moving
```
Comment (rarcifa) [inline Metaverse/Source/Metaverse/MetaverseCharacter.cpp:81]: space // Moving

File: Metaverse/Source/Metaverse/MetaverseCharacter.cpp [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+		EnhancedInputComponent->BindAction(MoveAction, ETriggerEvent::Triggered, this, &AMetaverseCharacter::Move);
+
+		//Looking
```
Comment (rarcifa) [inline Metaverse/Source/Metaverse/MetaverseCharacter.cpp:84]: space // Looking

File: Metaverse/Source/Metaverse/MetaverseCharacter.cpp [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+#include "Components/CapsuleComponent.h"
+#include "Components/InputComponent.h"
+#include "GameFramework/CharacterMovementComponent.h"
+#include "GameFramework/Controller.h"
+#include "GameFramework/SpringArmComponent.h"
+#include "EnhancedInputComponent.h"
+#include "EnhancedInputSubsystems.h"
+
+
+//////////////////////////////////////////////////////////////////////////
```
Comment (rarcifa) [inline Metaverse/Source/Metaverse/MetaverseCharacter.cpp:14]: is this needed?

File: Metaverse/Source/Metaverse/MetaverseCharacter.cpp [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+	if (APlayerController* PlayerController = Cast<APlayerController>(Controller))
+	{
+		if (UEnhancedInputLocalPlayerSubsystem* Subsystem = ULocalPlayer::GetSubsystem<UEnhancedInputLocalPlayerSubsystem>(PlayerController->GetLocalPlayer()))
+		{
+			Subsystem->AddMappingContext(DefaultMappingContext, 0);
+		}
+	}
+}
+
+//////////////////////////////////////////////////////////////////////////
```
Comment (rarcifa) [inline Metaverse/Source/Metaverse/MetaverseCharacter.cpp:69]: here as well?

File: Metaverse/Config/DefaultEngine.ini (hunk 1 above)
Comment (damoncro) [reply Metaverse/Config/DefaultEngine.ini:60]: It is auto generated by unreal, but ok to keep it empty.

File: Metaverse/Source/Metaverse.Target.cs (hunk 2 above)
Comment (damoncro) [reply Metaverse/Source/Metaverse.Target.cs:1]: Yes, unmodified files generated by unreal, copyright is correct at this moment.

File: Metaverse/Config/DefaultGame.ini (hunk 3 above)
Comment (damoncro) [reply Metaverse/Config/DefaultGame.ini:2]: It is a UID generated by https://docs.unrealengine.com/5.0/en-US/API/Runtime/Core/Misc/FGuid/. It should be fixed after project created. Let's leave it?

File: Metaverse/Source/Metaverse/MetaverseCharacter.cpp (hunk 8 above)
Comment (damoncro) [reply Metaverse/Source/Metaverse/MetaverseCharacter.cpp:69]: ok to remove

File: Metaverse/Source/Metaverse/MetaverseCharacter.cpp (hunk 7 above)
Comment (damoncro) [reply Metaverse/Source/Metaverse/MetaverseCharacter.cpp:14]: ok to remove

File: Metaverse/Source/Metaverse/MetaverseCharacter.cpp (hunk 4 above)
Comment (damoncro) [reply Metaverse/Source/Metaverse/MetaverseCharacter.cpp]: done

File: Metaverse/Source/Metaverse/MetaverseCharacter.cpp (hunk 5 above)
Comment (damoncro) [reply Metaverse/Source/Metaverse/MetaverseCharacter.cpp]: done

File: Metaverse/Source/Metaverse/MetaverseCharacter.cpp (hunk 6 above)
Comment (damoncro) [reply Metaverse/Source/Metaverse/MetaverseCharacter.cpp]: done

File: Metaverse/Config/DefaultEngine.ini (hunk 1 above)
Comment (damoncro) [reply Metaverse/Config/DefaultEngine.ini]: yes, all assets are from default template, free to use.

File: Metaverse/Config/DefaultGame.ini (hunk 3 above)
Comment (damoncro) [reply Metaverse/Config/DefaultGame.ini:2]: SecurityToken could be passed by 
```
UnrealAndroidFileTool.exe -s AB187923123CD123 -p com.OrganizationName.ProjectName shell -k CCEED369EB42CBA717F8528399C6F4E9
```

**Review by rarcif — APPROVED** [review]
lgtm

**Review by leejw51crypto — APPROVED** [review]
lgtm

**Pull request merged**